infrastructures/docker/src/main/java/org/eclipse/che/workspace/infrastructure/docker/DockerMachine.java

```diff
@@ -148,6 +150,14 @@ void setServerStatus(String serverRef, ServerStatus status) {
         server.setStatus(status);
     }
 
+    public void putResource(String targetPath, InputStream sourceStream) throws InfrastructureException {
+        try {
+            docker.putResource(PutResourceParams.create(container, targetPath, sourceStream));
+        } catch (IOException e) {
+            throw new InfrastructureException(e.getMessage(), e);
```

Comment:
exec method throws InfrastructureException it's why I decided to throw InfrastructureException here.
Is InternalInfrastructureException better here?